api/v1beta1/oscmachine_validation.go

```diff
@@ -0,0 +1,243 @@
+package v1beta1
+
+import (
+	"errors"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"regexp"
+	"strings"
+)
+
+const (
+	minIops = 0
+	maxIops = 13000
+	minSize = 0
+	maxSize = 14901
+)
+
+// ValidateOscMachineSpec validate each parameters of OscMachine spec
+func ValidateOscMachineSpec(spec OscMachineSpec) field.ErrorList {
+	var allErrs field.ErrorList
+	if spec.Node.Vm.KeypairName != "" {
+		if errs := ValidateVmKeypairName(spec.Node.Vm.KeypairName, field.NewPath("keypairName")); len(errs) > 0 {
+			allErrs = append(allErrs, errs...)
+		}
+	}
+	if spec.Node.Vm.ImageId != "" {
+		if errs := ValidateVmImageId(spec.Node.Vm.ImageId, field.NewPath("imageId")); len(errs) > 0 {
+			allErrs = append(allErrs, errs...)
+		}
+	}
+	if spec.Node.Vm.DeviceName != "" {
+		if errs := ValidateVmDeviceName(spec.Node.Vm.DeviceName, field.NewPath("deviceName")); len(errs) > 0 {
+			allErrs = append(allErrs, errs...)
+		}
+	}
+	if spec.Node.Vm.VmType != "" {
+		if errs := ValidateVmVmType(spec.Node.Vm.VmType, field.NewPath("vmType")); len(errs) > 0 {
```

Comment:
validateVmVmType ? 
(Vm repeated two times )

Reply:
change to nodeVm